### PR TITLE
Add liniarRemoveElement for Dlist

### PR DESCRIPTION
**Pull request by Darredevil**
https://issues.dlang.org/show_bug.cgi?id=9362

Followed the idea from https://github.com/dlang/phobos/pull/5784 and implemented the same for Dlist.

### CODE REVIEW

**Comment by Darredevil** [comment]
@andralex awaiting approval

File: std/container/dlist.d [hunk 1]
```diff
@@ -702,8 +702,46 @@ Complexity: $(BIGOH r.walkLength)
...
+
+        return false;
+    }
+
+
 private:
 
+    BaseNode* findNodeByValue(BaseNode* n, T value)
+    {
+        assert(n, "Error: un-initialized List");
```
Comment (andralex) [inline std/container/dlist.d]: shouldn't you just return null if the list is not initialized?

File: std/container/dlist.d [hunk 2]
```diff
@@ -766,6 +804,34 @@ private:
 {
     import std.algorithm.comparison : equal;
 
+    auto a = DList!int(-1, 1, 2, 1, 3, 4);
```
Comment (andralex) [inline std/container/dlist.d]: should include a test involving an empty list

File: std/container/dlist.d (hunk 1 above)
Comment (ZombineDev) [reply std/container/dlist.d]: I would also prefer to return null instead of asserting, though all the other `findNode*` functions do the same.

**Comment by Darredevil** [comment]
@andralex @ZombineDev should be ready for merge

File: std/container/dlist.d [hunk 3]
```diff
@@ -702,8 +702,46 @@ Complexity: $(BIGOH r.walkLength)
...
+
+Params:
+    value = value of the node to be removed
+
+Complexity: $(BIGOH n)
+     */
+    bool linearRemoveElement(T value)
+    {
+        auto n1 = findNodeByValue(_root, value);
+        // if node exists
```
Comment (andralex) [inline std/container/dlist.d]: delete, just repeats what the code says
